compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -1734,16 +1734,24 @@ object Objects:
       if cls.isAllOf(Flags.JavaInterface) then Bottom
       else evalType(tref.prefix, thisV, klass, elideObjectAccess = cls.isStatic)
 
+  val mutateErrorSet: mutable.Set[(ClassSymbol, ClassSymbol)] = mutable.Set.empty
   def errorMutateOtherStaticObject(currentObj: ClassSymbol, otherObj: ClassSymbol)(using Trace, Context) =
-    val msg =
-      s"Mutating ${otherObj.show} during initialization of ${currentObj.show}.\n" +
-      "Mutating other static objects during the initialization of one static object is forbidden. " + Trace.show
+    if mutateErrorSet.contains((currentObj, otherObj)) then ()
```

Comment:
The `add` method on `mutable.Set` combines checking whether the set contains an element (return `Boolean`) and adding that element if it doesn't. Replace `contains` and `+=` with `add`.